--- glibmm-2.34.1/gio/giomm/contenttype.cc.orig	2012-10-06 14:28:36.000000000 +0200
+++ glibmm-2.34.1/gio/giomm/contenttype.cc	2012-11-06 16:54:11.439593123 +0100
@@ -56,6 +56,7 @@
   return retvalue;
 }
 
+#ifndef G_OS_WIN32
 Glib::RefPtr<Gio::Icon> content_type_get_symbolic_icon(const Glib::ustring& type)
 {
   Glib::RefPtr<Icon> retvalue = Glib::wrap(g_content_type_get_symbolic_icon(type.c_str()));
@@ -63,6 +64,7 @@
     retvalue->reference(); //The function does not do a ref for us.
   return retvalue;
 }
+#endif
 
 bool content_type_can_be_executable(const Glib::ustring& type)
 {
